File: einterfaces/saml.go
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package einterfaces

import (
	"github.com/mattermost/mattermost-server/model"
)

type SamlInterface interface {
	ConfigureSP() error
	BuildRequest(relayState string) (*model.SamlAuthRequest, error)
	DoLogin(encodedXML string, relayState map[string]string) (*model.User, error)
	GetMetadata() (string, error)
}

File: webapp/channels/src/packages/types/src/saml.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type SamlCertificateStatus = {
    idp_certificate_file: string;
    private_key_file: string;
    public_certificate_file: string;
};

export type SamlMetadataResponse = {
    idp_descriptor_url: string;
    idp_url: string;
    idp_public_certificate: string;
};
